crates/sncast/src/starknet_commands/account/import.rs

```diff
@@ -98,36 +94,55 @@ pub async fn add(
         _ => fetched_class_hash,
     };
 
-    let legacy = check_if_legacy_contract(class_hash, add.address, provider).await?;
+    let chain_id = get_chain_id(provider).await?;
+    if let (Some(salt), Some(class_hash)) = (import.salt, class_hash) {
+        let sncast_account_type = match import.account_type {
+            AccountType::Argent => SNCastAccountType::Argent,
+            AccountType::Braavos => SNCastAccountType::Braavos,
+            AccountType::Oz => SNCastAccountType::OpenZeppelin,
+        };
+        let computed_address =
+            compute_account_address(salt, private_key, class_hash, sncast_account_type, chain_id);
```

Comment:
If we successfully fetched the class_hash for the given address doesn't that mean it's correct already? And getting there without `class_hash` is weird. Maybe this verification should be done at an earlier stage?

Reply:
I don't really understand what's the issue here 😅 We compute the account address and check if it's the same (as the one from user) right after resolving class hash which is needed for computation. Could you please explain what you mean?